deploy/api.yaml

```diff
@@ -0,0 +1,157 @@
+      x-amazon-apigateway-request-validators: "Validate body"
+      x-amazon-apigateway-integration:
+        type: "aws_proxy"
+        httpMethod: "POST"
+        uri:
+          Fn::Sub: "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${KBVSessionFunction.Arn}/invocations"
+        passthroughBehavior: "when_no_match"
+  /question:
+    get:
+      parameters:
+        - $ref: "#/components/parameters/SessionHeader"
+      responses:
+        "200":
+          description: "200 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Question"
+        "201":
+          description: "201 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Question"
+        "400":
+          description: "400 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: "500 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+      x-amazon-apigateway-request-validators: "Validate body"
+      x-amazon-apigateway-integration:
+        type: "aws_proxy"
+        httpMethod: "POST"
+        uri:
+          Fn::Sub: "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${KBVQuestionFunction.Arn}/invocations"
+        passthroughBehavior: "when_no_match"
+  /answer:
+    post:
+      parameters:
+        - $ref: "#/components/parameters/SessionHeader"
+      responses:
+        "201":
+          description: "201 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Answer"
+        "400":
+          description: "400 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: "500 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+      x-amazon-apigateway-request-validators: "Validate body"
+      x-amazon-apigateway-integration:
+        type: "aws_proxy"
+        httpMethod: "POST"
+        uri:
+          Fn::Sub: "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${KBVAnswerFunction.Arn}/invocations"
+        passthroughBehavior: "when_no_match"
+components:
+  parameters:
+    SessionHeader:
+      name: session-id
+      in: header
+      description: A UUID generated by the KBV API to act as a primary key for the KBVSessionTable in DynamoDB
+      required: true
+      schema:
+        type: string
+  schemas:
+    Session:
+      type: "object"
+      properties:
+        session-id:
+          type: "string"
+    Error:
+      title: "Error Schema"
+      type: "object"
+      properties:
```

Comment:
We should return to this when we look at the error flows, as we the frontend will almost always be redirecting back to core as part of the oauth flow, and the core system rendering the error pages.

As this redirect will have an error code in the param, as always we'll want to balance usability with security.

Reply:
Agreed